lib/publiccloud/ec2.pm

```diff
@@ -161,6 +161,7 @@ sub upload_boot_diagnostics {
     my ($self, %args) = @_;
     my $instance_id = $self->get_terraform_output('.vm_name.value[]');
     return if (check_var('PUBLIC_CLOUD_SLES4SAP', 1));
+    return unless (defined($instance_id));
```

Comment:
yes I think it make sense but let's be verbose here , let's add record_info complaining about missing instance_id

Reply:
done